### ISSUE TITLE
assign list presence tags to individual chemical records

### ISSUE THREAD

**Issue by kdionisio** [HumanExposure/factotum#937]
As a Factotum user
I want the ability to assign list presence tags to individual chemical records of type CP
So that I can assign keywords appropriately to individual chemical records within a single data document

Currently when list presence tags are assigned to chemicals in a CP document, the list presence tags are applied to uniformly to every chemical in the document, even though the data model allows for assigning the tags to individual chemicals.

Definition of Done:
1. ability to assign different list presence tags to individual chemical records within a single data document of group type CP 
2. maintain functionality to easily assign a tag across all chemical records in a data document if desired (e.g. if the document includes 200 chemical records that all require the same tag, this should still be a reasonably easy process)
3. will need to modify visual display on data document page of list presence tags assigned such that user can distinguish between tags which apply to every chemical record on the page and those which apply only to specific chemical records

**Comment by zachmunzenrider** [comment]
This will be tricky! Currently tags are updated for each chemical record via a brute-force loop over the entire set of chemicals, and it isn't clear to me how we'll keep everything in sync this way.

It almost makes me want to refactor the tags so that we have those that apply to the entire set (which under the hood applies to the List Presence itself) and those that apply specifically to the individual chemical.

When it comes time to display the tags associated with a specific chemical, we can show both kinds. This would also make it easier to style them differently, so users know which is which.

**Comment by mwfrost** [comment]
I agree that item 3 is the most challenging. If the only difference between per-ExtractedListPresence tags and per-ExtractedCPCat tags is whether or not the tag is applied to all the ELPs in an ExtractedCPCat, formatting them differently will involve checking with a query.

Could we make that item its own ticket?
